src/services/config_extraction.rs

```diff
@@ -286,9 +292,16 @@ impl ConfigExtractionService for ConfigExtractionServiceImpl {
         let large_distro_name = self
             .evg_config_utils
             .lookup_build_variant_expansion(LARGE_DISTRO_EXPANSION, build_variant);
+        let xlarge_distro_name = self
+            .evg_config_utils
+            .lookup_build_variant_expansion(XLARGE_DISTRO_EXPANSION, build_variant);
         let build_variant_name = build_variant.name.as_str();
 
-        if generated_task.use_large_distro() {
+        if generated_task.use_xlarge_distro() && xlarge_distro_name.is_some() {
```

Comment:
what if you specify both large and xlarge distros? Does that work and it looks like xlarge would take precedence?

Reply:
Yes, it works fine and xlarge takes precedence which I think is desired behavior.